pkgs/test/test/runner/github_reporter_test.dart

```diff
@@ -0,0 +1,346 @@
+          }, skip: true);''', '''
+          ::group::❎ skip test 1 (skipped)
+          ::endgroup::
+          ::group::❎ skip test 2 (skipped)
+          ::endgroup::
+          ::group::❎ skip test 3 (skipped)
+          ::endgroup::
+          🎉 0 tests passed, 3 skipped.''');
+    });
+
+    test('runs skipped tests along with successful tests', () {
+      return _expectReport('''
+          test('skip 1', () {}, skip: true);
+          test('success 1', () {});
+          test('skip 2', () {}, skip: true);
+          test('success 2', () {});''', '''
+          ::group::❎ skip 1 (skipped)
+          ::endgroup::
+          ::group::✅ success 1
+          ::endgroup::
+          ::group::❎ skip 2 (skipped)
+          ::endgroup::
+          ::group::✅ success 2
+          ::endgroup::
+          🎉 2 tests passed, 2 skipped.''');
+    });
+
+    test('runs skipped tests along with successful and failing tests', () {
+      return _expectReport('''
+          test('failure 1', () => throw TestFailure('oh no'));
+          test('skip 1', () {}, skip: true);
+          test('success 1', () {});
+          test('failure 2', () => throw TestFailure('oh no'));
+          test('skip 2', () {}, skip: true);
+          test('success 2', () {});''', '''
+          ::group::❌ failure 1 (failed)
+          oh no
+          test.dart 6:35  main.<fn>
+          ::endgroup::
+          ::group::❎ skip 1 (skipped)
+          ::endgroup::
+          ::group::✅ success 1
+          ::endgroup::
+          ::group::❌ failure 2 (failed)
+          oh no
+          test.dart 9:35  main.<fn>
+          ::endgroup::
+          ::group::❎ skip 2 (skipped)
+          ::endgroup::
+          ::group::✅ success 2
+          ::endgroup::
+          ::error::2 tests passed, 2 failed, 2 skipped.''');
+    });
+
+    test('displays the skip reason if available', () {
+      return _expectReport('''
+          test('skip 1', () {}, skip: 'some reason');
+          test('skip 2', () {}, skip: 'or another');''', '''
+          ::group::❎ skip 1 (skipped)
+          Skip: some reason
+          ::endgroup::
+          ::group::❎ skip 2 (skipped)
+          Skip: or another
+          ::endgroup::
+          🎉 0 tests passed, 2 skipped.''');
+    });
+  });
+
+  test('loadSuite, setUpAll, and tearDownAll hidden if no content', () {
+    return _expectReport('''
+          group('one', () {
+            setUpAll(() {/* nothing to do here */});
+            tearDownAll(() {/* nothing to do here */});
+            test('test 1', () {});
+          });''', '''
+          ::group::✅ one test 1
+          ::endgroup::
+          🎉 1 test passed.''');
+  });
+
+  test('setUpAll and tearDownAll show when they have content', () {
+    return _expectReport('''
+          group('one', () {
+            setUpAll(() { print('one'); });
+            tearDownAll(() { print('two'); });
+            test('test 1', () {});
+          });''', '''
+          ::group::⏺ one (setUpAll)
```

Comment:
Why not use ✅ for these as well? This looks to me like a warning indicator.

I would consider using this icon for skipped tests actually?

Reply:
There's some (incomplete) discussion of the icons here: https://github.com/dart-lang/test/pull/1678#issuecomment-1075677817. I did play around w/ a lot of different icons, including just editing the dom of the 'checks' page to see how things would look.

For this icon, I don't want to use the same icon as for a regular test; I think it's valuable for 6 users test ==> 6 test icons in the output. I'd like the additional test plumbing to have distinct icons from the passed tests. Additionally, the icon should be the same width as the other icons - so the LHS isn't ragged - and shouldn't use a color which users already attach some meaning to (red, yellow, ...).

Other similar unicode chars can be found here: https://unicode-table.com/en/blocks/dingbats/.

Reply:
What about ☑? That way it is different but still indicates success (it is a test that can fail). And it isn't a color that indicates a warning. My main complaint about the current choice is the orange color indicates a warning to me.

Reply:
I've been checking the rendering in-line in the test results, just so we're getting the font that github uses there. What I see for the current 'non-test' group char (⏺) is:

<img width="116" alt="Screen Shot 2022-03-29 at 9 45 09 AM" src="https://user-images.githubusercontent.com/1269969/160663421-91050ddb-aab3-4de5-9db5-dd444d6b4997.png">

Which - for me - is a gray color, not an orange color. I do agree that orange (or red, or yellow) would indicate a failure or problem of some kind to the user - something we should avoid unless there's an actual issue.

Here's what the ☑ char looks like in-line in test results (note the different size and style):

<img width="201" alt="Screen Shot 2022-03-29 at 9 40 14 AM" src="https://user-images.githubusercontent.com/1269969/160663127-b5af484e-502e-43b7-83a4-6bedb98c3d53.png">

Reply:
Oh weird... ya its orange on a chromebook :(. I am on my mac now and I see it as a greyish blue. These were both looking at the github UI, so the color is still different based on the platform and not the terminal.

That is pretty unfortunate. @munificent was suggesting using an older check mark symbol that is just the check mark (no background) and then coloring the check mark explicitly?

Reply:
I also think we can definitely launch this with any of the suggested choices, and revist later.

Reply:
> that is just the check mark (no background) and then coloring the check mark explicitly?

That would be great if we could do it :) From experimentation, we can't use ansi colors in the summary line of a group here.

> I also think we can definitely launch this with any of the suggested choices, and revist later.

Let's do that then; I don't think there's an idea set of icons unfortunately - I've experimented w/ many. I'll change to the green checkmark, and we can iterate as needed.